### PR TITLE
Adds wiki subscribe button on wiki action bar

### PR DESCRIPTION
**Pull request by Royal-lobster**
Closes https://github.com/EveripediaNetwork/issues/issues/1009
Closes https://github.com/EveripediaNetwork/issues/issues/1004

# Changes
- adds wiki subscribe button on wiki action bar
- makes useProfileData more typesafe data
- adds wiki subscription modal to show the current subscription status and to toggle subscription to wiki
- fixes about page features background in dark mode
- Adds tooltips to disabled action buttons 

# Screenshots

https://user-images.githubusercontent.com/52039218/216313640-ff48a8ce-0862-45bc-a22a-9e38405d069d.mp4

## On USER_TOKEN absense

https://user-images.githubusercontent.com/52039218/216319519-d0a2e163-1d02-4628-a34d-2496b987de0f.mp4

### CODE REVIEW

**Comment by kesar** [comment]
mmm great idea!

**Review by Simplypheyie96 — APPROVED** [review]
It looks cool. can see you already implemented the inactive state.

@Softdev1 please create an issue for some tooltip guide/message for the inactive state buttons on wiki page.  so we can design and implement that.

File: src/components/Settings/Notification/SearchWikiNotifications.tsx [hunk 1]
```diff
@@ -117,9 +120,10 @@ const SearchWikiNotifications = () => {
...
 
   const inputRef = useRef<HTMLInputElement | null>(null)
-  const { profileData } = useUserProfileData(getUserAddressFromCache(), {
-    withAllSettings: true,
-  })
+  const { profileData } = useUserProfileData(
+    UserProfileFetchOptions.WITH_ALL_SETTINGS,
+    getUserAddressFromCache(),
+  )
   const router = useRouter()
```
Comment (Adebesin-Cell) [inline src/components/Settings/Notification/SearchWikiNotifications.tsx:127]: Same with this file > 200, the useEventListener logic is used twice, on Navbar and here, we can move it into one common file and use.

**Review by Adebesin-Cell — CHANGES_REQUESTED** [review]

File: src/components/Notification/NotificationCard.tsx [hunk 2]
```diff
@@ -104,9 +107,9 @@ const NotificationCard = ({
   defaultSubscribed,
 }: NotificationCardProps) => {
   const userAddress = getUserAddressFromCache() as string
-  const { setAccount, profileData } = useUserProfileData('', {
-    withAllSettings: true,
-  })
+  const { setAccount, profileData } = useUserProfileData(
+    UserProfileFetchOptions.WITH_ALL_SETTINGS,
+  )
   const isWikiSubscribed = useIsWikiSubscribed(wikiId, userAddress)
```
[the remark this answers is not in the document]
Comment (Royal-lobster) [reply src/components/Notification/NotificationCard.tsx:113]: Good catch ! I guess we can add this to a new issue cuz it is unrelated to this pr

File: src/components/Settings/Notification/SearchWikiNotifications.tsx (hunk 1 above)
Comment (Royal-lobster) [reply src/components/Settings/Notification/SearchWikiNotifications.tsx:127]: Same, We can address this in another refactoring pr

File: src/components/About/AboutFeaturesCard.tsx [hunk 3]
```diff
@@ -13,7 +13,14 @@ const AboutFeaturesCard = ({
   content,
   icon,
 }: AboutFeaturesCardProps) => (
-  <Box bgColor="pageBg" p={8} rounded="md">
+  <Box
+    bgColor="gray.100"
+    _dark={{
```
[the remark this answers is not in the document]
Comment (Royal-lobster) [reply src/components/About/AboutFeaturesCard.tsx:18]: Oh yes ! Will add a commit for this. Thanks!

File: src/components/Notification/NotificationCard.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Adebesin-Cell) [reply src/components/Notification/NotificationCard.tsx:113]: Yeah 👍, @Softdev1 You can an issue to refactor the code on Notifications.

**Review by Adebesin-Cell — APPROVED** [review]
Awesome 🚀.

**Comment by Adebesin-Cell** [comment]
> i would suggest to avoid the remove notification as a primary button in the dialog. We use primary colors when we want to guide user to take a decision. I dont think we want the user to unsubscribe after subscribe. 😅
> 
> does the button change if user is already subscribed?
> 
> also make sure it works for mobile (not big priority, but that can work at least) 👍👍

Haha 😅, yeah. Thanks for pointing that out @kesar.

**Comment by Royal-lobster** [comment]
> 

Good point ! i'll change the unsubscribe to outline button instead i think that should work. Yes the button changes to "unsubscribe" when it is already subscribed. It works for mobile too !

**Comment by Adebesin-Cell** [comment]
Great 👍.

**Review by Softdev1 — APPROVED** [review]

**Pull request merged**